subprojects/build-cache/src/main/java/org/gradle/caching/internal/BuildCacheHasher.java

```diff
@@ -29,5 +34,17 @@
     BuildCacheHasher putBoolean(boolean b);
     BuildCacheHasher putString(CharSequence charSequence);
     BuildCacheHasher putNull();
+
+    /**
+     * Marks this build cache hash as invalid.
+     */
+    void markAsInvalid();
+
+    /**
+     * Returns the combined hash.
+     *
+     * If the build cache hash is invalid, {@code null is returned}.
+     */
+    @Nullable
     HashCode hash();
```

Comment:
I wonder if we should do it like this, or if we should have this method throw something like `IllegalStateException` and instead have an `isValid()` method? I'd rather deal with descriptive exceptions than `null`s.

Reply:
I changed to use an `isValid` method.